src/controllers/VueController.php

```diff
@@ -115,35 +97,29 @@ private function fetchOrders($id = null) : array
             $orders->where(['paidStatus' => strtolower($paymentType)]);
         }
 
-        $result['previousPeriod'] = $orders->dateOrdered(['and', ">= {$newStart}", "< {$currentStart}"])->all();
-        $result['currentPeriod']  = $orders->dateOrdered(['and', ">= {$currentStart}", "< {$newEnd}"])->all();
-
-        return $result;
-    }
+        if($id) {
+            $product = Variant::find()->id($id)->one();
+            $orders->hasPurchasables([$product]);
+        }
 
-    /**
-     * Get all products that are in the system.
-     *
-     * @return array - All products, or an empty array
-     */
-    private function fetchProducts() : array
-    {
-        return Product::find()->anyStatus()->all() ?: [];
-    }
+        $result = [
+            'previousPeriod' => $orders->dateOrdered(['and', ">= {$newStart}", "< {$currentStart}"])->all(),
+            'currentPeriod'  => $orders->dateOrdered(['and', ">= {$currentStart}", "< {$newEnd}"])->all()
+        ];
```

Comment:
RE https://github.com/FosterCommerce/craft-commerceinsights/pull/14#discussion_r674909598

> 2 queries is less efficient than one because they can't be run in parallel. Since there won't be separate get-stats and get-orders endpoints in the future, instead only 1 API call, I'd rather keep this as-is. Removed $this->orders

I wasn't aware the goal would be to merge the two endpoints.

Consider a result set which includes 1,000 orders, 10,000 or 100,000 orders. Each of those results sets could be doubled to account for the previous period. 

- Should users' servers be able to handle processing of that much data on every request?
- Where will paging happen? In other words, will you fetch the entire set of data and page it in code, or will that happen in SQL?
- If paging happens in SQL, how will you generate the stats?
- How would paging be implemented for the previous period if that data isn't supposed to be rendered other than as stats?

Something else to consider here: 

> 2 queries is less efficient than one because they can't be run in parallel

That is true for php. However,  PHP FPM can be configured to run requests in parallel using child processes, see process management config https://www.php.net/manual/en/install.fpm.configuration.php#pm. AFAIK Apache does a similar thing.

Wouldn't it be more efficient to allow the client to make multiple requests for the data? One request for a small amount of stats data, and another for paged order data?

Reply:
Well, if it's not combined, then it'll have to fetch 50,000 orders to show the table, plus 100,000 more orders in a separate API call and query to show the stats. If it's combined, it can just fetch 100,000 in one API call and one query.

Reply:
@clowerweb - I don't think you've answered Ty's questions.